### PR TITLE
#190, #200 :: Migrate to new cc provider.

### PR DESCRIPTION
**Pull request by mfarrugi**
Locally tested this works with and without the flag on bazel v22.

### CODE REVIEW

**Comment by mfarrugi** [comment]
cc @hlopko @damienmg

File: rust/private/legacy_cc_starlark_api_shim.bzl [hunk 1]
```diff
@@ -0,0 +1,28 @@
+"""
+Utility functions that replace old C++ Starlark API using the new API.
+See migration instructions in https://github.com/bazelbuild/bazel/issues/7036
+
+Pasted from https://gist.github.com/oquenchil/7e2c2bd761aa1341b458cc25608da50c
```
Comment (damienmg) [inline rust/private/legacy_cc_starlark_api_shim.bzl]: There is no license from this code, can we get the author to add one?

Comment (hlopko) [reply rust/private/legacy_cc_starlark_api_shim.bzl]: Author is on vacation, and I'm pretty sure attribution is not needed (in the same way as StackOverflow answers aren't licensed).

Comment (mfarrugi) [reply rust/private/legacy_cc_starlark_api_shim.bzl]: StackOverflow answers are licensed as creative commons. If this not covered by some Google default oss license?

Comment (hlopko) [reply rust/private/legacy_cc_starlark_api_shim.bzl]: Ok, then I'm not a lawyer :) So don't copypaste, write it by your own, and don't attribute?

**Pull request merged**